torch/jit/_monkeytype_config.py

```diff
@@ -72,6 +73,10 @@ def consolidate_types(self, qualified_name: str) -> Dict:
                     if inspect.getmodule(_type) == typing:
                         _type_to_string = str(_type)
                         _all_type += _type_to_string.replace('typing.', '') + ','
+                    elif inspect.getmodule(_type) == torch.nn.parameter:
```

Comment:
Why does `nn.Parameter` need special handling?

It should be handled here: https://github.com/pytorch/pytorch/blob/7f2e6201052fa910e845cd2e3555c486e3cce3b9/torch/jit/annotations.py#L280

Reply:
The `if` condition added here covers all classes inside `torch.nn.parameter` module (aka. `torch/nn/parameter.py`, instead of just `torch.nn.Parameter` class. Note their names are different (lower vs upper case). Is this intended?

Reply:
The fully qualified name of `nn.Parameter` is `torch.nn.parameter.Parameter`. So earlier only `_type.name` (`Parameter`) was used to infer the annotation for the argument while using monkeytype and torchscript failed to understand. Now after using the fully qualified name , TS doesn't complain.

Reply:
I see, I thought `_type.name` contains full qualified name. If this is the case, that seems like a more universal problem.

I wonder why this issue doesn't show up in our torchbench experiment where a model could use a class defined in other (non-main) modules.